### PR TITLE
lib: bin: lwm2m_carrier: update to version 0.9.0

### PR DESCRIPTION
**Pull request by nordic-hani**
Update library to version 0.9.0.

Signed-off-by: Håvard Vermeer <havard.vermeer@nordicsemi.no>

### CODE REVIEW

File: lib/bin/lwm2m_carrier/include/lwm2m_carrier.h [hunk 1]
```diff
@@ -9,11 +9,6 @@
 #include <stdint.h>
 #include <stddef.h>
 
-#ifdef __cplusplus
```
Comment (nordic-hani) [inline lib/bin/lwm2m_carrier/include/lwm2m_carrier.h:12]: Will add back the c++ guard.

**Comment by nordic-hani** [comment]
Forgot to format style for NCS. Will fix the checkpath issues.

File: lib/bin/lwm2m_carrier/doc/CHANGELOG.rst [hunk 2]
```diff
@@ -5,6 +5,34 @@ Changelog
...
+* Expanded API with ``certification_mode`` variable that chooses between certification or test servers upon initialization the LwM2M carrier library.
+* Expanded API with ``apn`` variable to set a custom APN upon initialization the LwM2M carrier library.
+* Added Kconfig configuration "LWM2M_CARRIER_USE_CUSTOM_BOOTSTRAP_PSK". PSK Key is now set independantly of custom URI.
+* PSK format has changed to be more copy paste friendly.
+  * Was byte array, for example ``static const char bootstrap_psk[] = {0x01, 0x02, 0xab, 0xcd, 0xef};``
+  * Is now a null-terminated string that must be composed of hexadecimal numbers. For example ``static const char bootstrap_psk = "0102abcdef";``.
+* Added lwm2m_carrier_event_deferred_t to retrieve the event reason and timeout.
+* Added FOTA errors to LWM2M_CARRIER_EVENT_ERROR event.
+
+
```
Comment (metc) [inline lib/bin/lwm2m_carrier/doc/CHANGELOG.rst:35]: Same "Known issues and limitations" as before regarding DTLS/TLS sessions that can be run in parallel. This should be maybe mentioned (same as below for the previous versions).

File: lib/bin/lwm2m_carrier/doc/CHANGELOG.rst [hunk 3]
```diff
@@ -5,6 +5,34 @@ Changelog
...
+Certification status
+====================
+
+The library is not certified with any carrier.
+
+Changes
+=======
+
+* Added new APIs to support portfolio objects. A new portfolio object instance can be created using ``lwm2m_carrier_portfolio_instance_create()``.
```
Comment (kacperradoszewski) [inline lib/bin/lwm2m_carrier/doc/CHANGELOG.rst:23]: There is only one object, but multiple instances Maybe rephrase the first sentence to something like: `Added new APIs to create and access Portfolio object instances`? I should have changed this earlier, sorry.

File: lib/bin/lwm2m_carrier/doc/CHANGELOG.rst [hunk 4]
```diff
@@ -5,6 +5,34 @@ Changelog
...
+  ``lwm2m_carrier_identity_read()`` and ``lwm2m_carrier_identity_write()`` are used to read and write to corresponding Identity resource fields of
+  a given Portfolio object instance.
+* Expanded API with ``certification_mode`` variable that chooses between certification or test servers upon initialization the LwM2M carrier library.
```
Comment (kacperradoszewski) [inline lib/bin/lwm2m_carrier/doc/CHANGELOG.rst:26]: I assume it's `the initialization of the`

File: lib/bin/lwm2m_carrier/doc/CHANGELOG.rst [hunk 5]
```diff
@@ -5,6 +5,34 @@ Changelog
...
+* Expanded API with ``apn`` variable to set a custom APN upon initialization the LwM2M carrier library.
```
Comment (kacperradoszewski) [inline lib/bin/lwm2m_carrier/doc/CHANGELOG.rst:27]: I assume it's `the initialization of the`

File: lib/bin/lwm2m_carrier/doc/CHANGELOG.rst [hunk 6]
```diff
@@ -5,6 +5,34 @@ Changelog
...
+* Added Kconfig configuration "LWM2M_CARRIER_USE_CUSTOM_BOOTSTRAP_PSK". PSK Key is now set independently of custom URI.
```
Comment (kacperradoszewski) [inline lib/bin/lwm2m_carrier/doc/CHANGELOG.rst:28]: PSK stands for Pre-Shared Key, so no need for `Key`

**Comment by kacperradoszewski** [comment]
We probably need to fix the `WARNING:LONG_LINE_COMMENT: line over 80 characters` checkpatch warnings

File: lib/bin/lwm2m_carrier/doc/CHANGELOG.rst [hunk 7]
```diff
@@ -5,6 +5,40 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+liblwm2m_carrier 0.9.0
+**********************
+
+Snapshot release for modem firmware version 1.2.0 and the |NCS| 1.3.0.
+
+This release is intended to let users begin integration towards the AT&T and Verizon device management platforms.
+It can be used for development and testing only. It's not ready for certification.
```
Comment (umapraseeda) [inline lib/bin/lwm2m_carrier/doc/CHANGELOG.rst:14]:
```suggestion
It can be used for development and testing only. 
It is not ready for certification.
```

File: lib/bin/lwm2m_carrier/doc/CHANGELOG.rst [hunk 8]
```diff
@@ -5,6 +5,40 @@ Changelog
...
+
+The library is not certified with any carrier.
+
+Changes
+=======
+
+* Added new APIs to create and access portfolio object instances.
+  A new portfolio object instance can be created using ``lwm2m_carrier_portfolio_instance_create()``.
+  ``lwm2m_carrier_identity_read()`` and ``lwm2m_carrier_identity_write()`` are used to read and write to the corresponding Identity resource fields of
+  a given portfolio object instance.
```
Comment (umapraseeda) [inline lib/bin/lwm2m_carrier/doc/CHANGELOG.rst:27]: Spacing
```suggestion
  ``lwm2m_carrier_identity_read()`` and ``lwm2m_carrier_identity_write()`` are used to read and write to the corresponding Identity resource fields of a given portfolio object instance.
```

File: lib/bin/lwm2m_carrier/doc/CHANGELOG.rst [hunk 9]
```diff
@@ -5,6 +5,40 @@ Changelog
...
+* Expanded API with "certification_mode" variable that chooses between certification or test servers upon the initialization of the LwM2M carrier library.
+* Expanded API with "apn" variable to set a custom APN upon the initialization of the LwM2M carrier library.
+* Added the LWM2M_CARRIER_USE_CUSTOM_BOOTSTRAP_PSK Kconfig option.
+  PSK Key is now set independently of custom URI.
```
Comment (umapraseeda) [inline lib/bin/lwm2m_carrier/doc/CHANGELOG.rst:31]: Missing Bullet?
```suggestion
  * PSK Key is now set independently of custom URI.
```

File: lib/bin/lwm2m_carrier/doc/CHANGELOG.rst [hunk 10]
```diff
@@ -5,6 +5,41 @@ Changelog
...
```
Comment (kacperradoszewski) [inline lib/bin/lwm2m_carrier/doc/CHANGELOG.rst:32]: `Pre-Shared Key` or just `PSK`

File: lib/bin/lwm2m_carrier/doc/CHANGELOG.rst [hunk 11]
```diff
@@ -5,6 +5,41 @@ Changelog
...
+=======
+
+* Added new APIs to create and access portfolio object instances.
+  A new portfolio object instance can be created using ``lwm2m_carrier_portfolio_instance_create()``.
+  ``lwm2m_carrier_identity_read()`` and ``lwm2m_carrier_identity_write()`` are used to read and write to the corresponding Identity resource fields of
+  a given portfolio object instance.
+* Expanded API with "certification_mode" variable that chooses between certification or test servers upon the initialization of the LwM2M carrier library.
+* Expanded API with "apn" variable to set a custom APN upon the initialization of the LwM2M carrier library.
+* Added the LWM2M_CARRIER_USE_CUSTOM_BOOTSTRAP_PSK Kconfig option.
+  * PSK Key is now set independently of custom URI.
```
Comment (umapraseeda) [inline lib/bin/lwm2m_carrier/doc/CHANGELOG.rst:32]:
```suggestion
* Added the LWM2M_CARRIER_USE_CUSTOM_BOOTSTRAP_PSK Kconfig option.

  * PSK Key is now set independently of custom URI.
```

File: lib/bin/lwm2m_carrier/doc/CHANGELOG.rst [hunk 12]
```diff
@@ -5,6 +5,42 @@ Changelog
...
+* Expanded API with "certification_mode" variable that chooses between certification or test servers upon the initialization of the LwM2M carrier library.
+* Expanded API with "apn" variable to set a custom APN upon the initialization of the LwM2M carrier library.
+* Added the LWM2M_CARRIER_USE_CUSTOM_BOOTSTRAP_PSK Kconfig option.
+
+  * PSK Key is now set independently of custom URI.
+
+* PSK format has been modified to be more user-friendly.
+
+  * Previous format: Byte array. For example, ``static const char bootstrap_psk[] = {0x01, 0x02, 0xab, 0xcd, 0xef};``.
+  * Current format: A null-terminated string that must be composed of hexadecimal numbers.
```
Comment (stig-bjorlykke) [inline lib/bin/lwm2m_carrier/doc/CHANGELOG.rst:37]: May need to add a reference to the Kconfig where this is set.

File: samples/nrf9160/lwm2m_carrier/src/main.c [hunk 13]
```diff
@@ -57,9 +67,13 @@ int lwm2m_carrier_event_handler(const lwm2m_carrier_event_t *event)
 	case LWM2M_CARRIER_EVENT_READY:
 		printk("LWM2M_CARRIER_EVENT_READY\n");
 		break;
-	case LWM2M_CARRIER_EVENT_DEFERRED:
+	case LWM2M_CARRIER_EVENT_DEFERRED: {
+		const lwm2m_carrier_event_deferred_t *devt = event->data;
 		printk("LWM2M_CARRIER_EVENT_DEFERRED\n");
+		printk("Reason %u, timeout %d seconds\n",
```
Comment (stig-bjorlykke) [inline samples/nrf9160/lwm2m_carrier/src/main.c:73]: You should probably print the reason text here, like we do for error code.

File: lib/bin/lwm2m_carrier/os/lwm2m_carrier.c [hunk 14]
```diff
@@ -30,8 +30,6 @@ void lwm2m_carrier_thread_run(void)
 
 #ifdef CONFIG_LWM2M_CARRIER_USE_CUSTOM_BOOTSTRAP_URI
 	config.bootstrap_uri = CONFIG_LWM2M_CARRIER_CUSTOM_BOOTSTRAP_URI;
-#endif
-#ifdef CONFIG_LWM2M_CARRIER_USE_CUSTOM_BOOTSTRAP_PSK
 	config.psk = (char *)bootstrap_psk;
```
Comment (stig-bjorlykke) [inline lib/bin/lwm2m_carrier/os/lwm2m_carrier.c:33]: The format has changed so you should create a new Kconfig for the bootstrap PSK.

**Comment by lemrey** [comment]
The doc PR is going in first: https://github.com/nrfconnect/sdk-nrf/pull/2194
